File: 01FlutterApps/middle_layer_springboot/knpiano_java/src/main/java/com/liu/springboot04web/bean/KnSub001Bean.java
package com.liu.springboot04web.bean;

import java.math.BigDecimal;
import java.security.Timestamp;
import java.util.Date;

import org.springframework.format.annotation.DateTimeFormat;

public class KnSub001Bean implements BzlFudousanBean{
    private String subjectId;
    private String subjectName;
    private BigDecimal subjectPrice;
    private Integer delFlg;
    //@DateTimeFormat(pattern = "yyyy/MM/dd")
    private Date createDate;

    //@DateTimeFormat(pattern = "yyyy/MM/dd")
    private Date updateDate;

    public String getSubjectId() {
        return subjectId;
    }
    public void setSubjectId(String subjectId) {
        this.subjectId = subjectId;
    }
    public String getSubjectName() {
        return subjectName;
    }
    public void setSubjectName(String subjectName) {
        this.subjectName = subjectName;
    }
    public BigDecimal getSubjectPrice() {
        return subjectPrice;
    }
    public void setSubjectPrice(BigDecimal subjectPrice) {
        this.subjectPrice = subjectPrice;
    }
    public Integer getDelFlg() {
        return delFlg;
    }
    public void setDelFlg(Integer delFlg) {
        this.delFlg = delFlg;
    }
    public Date getCreateDate() {
        return createDate;
    }
    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }
    public Date getUpdateDate() {
        return updateDate;
    }
    public void setUpdateDate(Date updateDate) {
        this.updateDate = updateDate;
    }
}

File: 01FlutterApps/front_dart_flutter/lib/01LessonMangement/2SubjectManagement/KnSub001Bean.dart
class KnSub001Bean {
  final String subjectId;
  final String subjectName;
  final double subjectPrice;
  final int delFlg;

  KnSub001Bean( {
    required this.subjectId, 
    required this.subjectName, 
    required this.subjectPrice,
    required this.delFlg,
    });

  factory KnSub001Bean.fromJson(Map<String, dynamic> json) {
    return KnSub001Bean(
      subjectId: json['subjectId'],
      subjectName: json['subjectName'],
      subjectPrice: json['subjectPrice'].toDouble(),
      delFlg: json['delFlg'] ?? 0,
    );
  }
}
